django/middleware/common.py

```diff
@@ -113,7 +113,8 @@ def process_response(self, request, response):
             if self.should_redirect_with_slash(request):
                 return self.response_redirect_class(self.get_full_path_with_slash(request))
 
-        if settings.USE_ETAGS:
+        need_etag = 'no-store' not in response.get('Cache-Control', '').split(',')
```

Comment:
This doesn't account for spaces around the comma, or for varying case. Using the [`has_vary_header()`](https://github.com/django/django/blob/ac77c55bc5fc54cd763a7ae426784650a8cc97c9/django/utils/cache.py#L254) model, I would import `cc_delim_re` and use that. Something like (untested):

    cc_headers = cc_delim_re.split(response.get('Cache-Control', ''))
    need_etag = 'no-store' not in (cc.lower() for cc in cc_headers)